### PR TITLE
Add --list-extensions command

### LINKED ISSUE #3303: New CLI option: --list-extensions
Right now it's very hard to tell what extensions are available. An extension that is not discoverable might as well be thrown in the trash, so add a command to list them, analogous to `--list-msgs`.

Similar to, but not as extreme as, https://github.com/PyCQA/pylint/issues/970.

### PR DESCRIPTION
**Pull request by anubh-v**
## Steps

- [ ] Add yourself to CONTRIBUTORS if you are a new contributor.
- [X] Add a ChangeLog entry describing what your PR does.
- [X] If it's a new feature or an important bug fix, add a What's New entry in `doc/whatsnew/<current release.rst>`.
- [X] Write a good description on what the PR does.

## Description

This PR adds a CLI command `--list-extensions` that lists the names of all extensions
present in `pylint.extensions`, as well as a short description of each extension.
Each extension's description is taken from the corresponding module's docstring.

## Type of Changes

|   | Type |
| ------------- | ------------- |
| ✓  | :sparkles: New feature |

## Related Issue 
Closes #3303

### CODE REVIEW

**Comment by anubh-v** [comment]
@PCManticore, I've done a first draft of this new command. May I get your feedback on it?

This implementation of `--list-extensions` will display the name of each extension under `pylint.extensions` as well as a description of each extension.

The description for each extension is taken from the extension module's docstring.

File: pylint/lint.py [hunk 1]
```diff
@@ -1843,6 +1861,14 @@ def cb_verbose_mode(self, *args, **kwargs):
         self.verbose = True
 
 
+def cb_list_extensions(option, optname, value, parser):
+    """List all the extensions under pylint.extensions"""
+    for extension in extensions.extension_names:
+        print(extension)
+        print(extensions.get_extension_documentation(extension))
```
Comment (PCManticore) [inline pylint/lint.py]: This command should just list the extensions without the documentation. The documentation can be accessed on the documentation website or we can add a separate command to grab the entire documentation of a plugin, but right now I suggest to remove this line.

**Comment by anubh-v** [comment]
@PCManticore, thanks for taking the time to review this. I will make the changes shortly.

**Comment by anubh-v** [comment]
@PCManticore, thanks for taking the time to review this. I've made the suggested changes.
Please let me know if this can be improved further.

**Comment by anubh-v** [comment]
@PCManticore, are there any further improvements you would like for this command? Thanks again for reviewing this

**Comment by anubh-v** [comment]
@PCManticore, do let me know if you would like further improvements on this PR.
Thanks very much for reviewing and giving us contributors a chance to help develop pylint!

**Comment by PCManticore** [comment]
Hi @anubh-v I just merged this locally to get rid of the conflicts. Thank you again for the contribution, it's much appreciated!

**Pull request closed without merging**